### PR TITLE
Remove backward compatible ADDED event from loadAll.

### PR DESCRIPTION
**Pull request by ahmetmircik**
addressed https://github.com/hazelcast/hazelcast/pull/13501#discussion_r206501299

### CODE REVIEW

File: hazelcast/src/test/java/com/hazelcast/journal/AbstractEventJournalBasicTest.java [hunk 1]
```diff
@@ -221,13 +218,7 @@ public void receiveLoadedEventsWhenLoadAll() throws Exception {
         testLoadAll(LOADED);
     }
 
-    @Test
-    public void receiveAddedEventsWhenLoadAll() throws Exception {
-        testLoadAll(ADDED);
-    }
-
     private void testLoadAll(EventType expectedEventType) throws Exception {
```
Comment (blazember) [inline hazelcast/src/test/java/com/hazelcast/journal/AbstractEventJournalBasicTest.java]: Think this method is not needed anymore and it's content can be moved into `receiveLoadedEventsWhenLoadAll()`.

Comment (ahmetmircik) [reply hazelcast/src/test/java/com/hazelcast/journal/AbstractEventJournalBasicTest.java]: done.

File: hazelcast/src/test/java/com/hazelcast/journal/AbstractEventJournalBasicTest.java [hunk 2]
```diff
@@ -73,7 +72,6 @@
     private int partitionId;
     private TruePredicate<EJ_TYPE> TRUE_PREDICATE = new TruePredicate<EJ_TYPE>();
     private Function<EJ_TYPE, EJ_TYPE> IDENTITY_FUNCTION = new IdentityFunction<EJ_TYPE>();
-    private boolean loadAllPublishesAdded = false;
 
     private void init() {
```
Comment (blazember) [inline hazelcast/src/test/java/com/hazelcast/journal/AbstractEventJournalBasicTest.java]: We can add the `@Before` annotation back here and remove the `init()` calls from the test cases. I introduced explicit `init()` calls only to make the load behavior parameterized, see https://github.com/hazelcast/hazelcast/pull/13674/files#r214365582.

Comment (ahmetmircik) [reply hazelcast/src/test/java/com/hazelcast/journal/AbstractEventJournalBasicTest.java]: lately seen this one, also addressed this.

**Pull request merged**